src/drivers/intel/cavs/ssp.c

```diff
@@ -365,12 +414,29 @@ static int ssp_set_config(struct dai *dai,
 		if (ssp_freq[clk_index].enc != CLOCK_SSP_XTAL_OSCILLATOR)
 			sscr0 |= SSCR0_ECS;
 	} else {
-		trace_ssp_error("ssp_set_config() error: BCLK %d",
-				config->ssp.bclk_rate);
-		ret = -EINVAL;
-		goto out;
-	}
+		/* check if we can get target BCLK with M/N */
+		for (i = 0; i <= MAX_SSP_FREQ_INDEX; i++) {
+			if (find_mn(ssp_freq[i].freq, config->ssp.bclk_rate,
+				    &mdiv, &i2s_m, &i2s_n)) {
+				clk_index = i;
+				break;
+			}
+		}
+
+		if (clk_index < 0) {
+			trace_ssp_error("ssp_set_config() error: BCLK %d",
```

Comment:
BCLK = :)

Reply:
Well, error is caused by invalid BCLK. Previously it would fail just if couldn't find valid SCR, now it can get requested BCLK with M/N also, so if you fail here it means, it failed in both.

Reply:
I mean just the string should include "=", just like in other traces "M = %d"   :D

Reply:
Error messages in this file have no `=`, only debug/info traces have `something = value`

Reply:
oh, ok then, so we have still some differences in trace formats.